sky/clouds/kubernetes.py

```diff
@@ -410,6 +423,14 @@ def make_deploy_resources_variables(
             ('kubernetes', 'provision_timeout'),
             10,
             override_configs=resources.cluster_config_overrides)
+
+        # Set environment variables for the pod. Note that SkyPilot env vars
+        # are set separately when the task is run. These env vars are
+        # independent of the SkyPilot task to be run.
+        k8s_env_vars = {
+            kubernetes_utils.IN_CLUSTER_CONTEXT_NAME_ENV_VAR: context
+        }
```

Comment:
I am not super sure of using this env var for passing around the context just for looking up the context in the remote_identity dict (correct me if I am wrong). 

If this is mainly for looking up a remote identity for jobs/serve controller, it might be better to just replace the key in `remote_identity` to `in-cluster` instead, like the one we did here: https://github.com/skypilot-org/skypilot/blob/f2991b144d4b15eac55dd7f759f361b6146033b3/sky/utils/controller_utils.py#L600-L626

Reply:
Other than looking up `remote_identity`, this envvar is also required to maintain parity with the region<->context mapping that we have created for SkyPilot.

E.g., if a user defines a YAML like:
```
resources:
  cloud: kubernetes
  region: gke_sky-dev-465_us-central1-c_gkeusc6
```

This YAML does not work on `sky jobs launch` on current master:
```
(sky-3ff3-romilb, pid=1481) ValueError: Context gke_sky-dev-465_us-central1-c_gkeusc6 not found in kubeconfig. Kubernetes only supports context names as regions. Available contexts: ['in-cluster']
```

With this PR, we can have region map to the right context consistently, even if it may be running in-cluster.